src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,963 @@
+                Next
+                FileList = FileSystem.GetFiles(TestBase.TestDirectory, SearchOption.SearchTopLevelOnly, "*.vb")
+                Assert.True(FileList.Count = 3)
+                For i As Integer = 0 To 2
+                    Assert.True(TestFileList.Contains(FileList(i)))
+                Next
+                Dim NewSubDirectoryPath As String = IO.Path.Combine(TestBase.TestDirectory, "GetFiles_Directory_SearchOption_WildcardsNewSubDirectory")
+                IO.Directory.CreateDirectory(NewSubDirectoryPath)
+                TestFileList.Add(CreateTestFile(TestBase, SourceData, PathFromBase:="GetFiles_Directory_SearchOption_WildcardsNewSubDirectory", TestFileName:="NewFile.cs"))
+                FileList = FileSystem.GetFiles(TestBase.TestDirectory, SearchOption.SearchAllSubDirectories, "*.cs")
+                Assert.True(FileList.Contains(TestFileList.Last), "File in Subdirectory not found")
+                Assert.True(FileList.Count = 4, $"4 files expected, {FileList.Count} returned from FileSystem.GetFiles")
+            End Using
+        End Sub
+
+        <Fact>
+        Public Sub LongDirectoryPathTest()
+            Using TestBase As New FileIOTests
+                Dim PathLength As Integer = TestBase.TestDirectory().Length
+                Assert.True(PathLength < 257) ' Need room for slash and new directory name
+                Dim DirectoryName As String = New String("A"c, 30)
+
+                Assert.True(DirectoryName.Length < 248, $"DirectoryBaseName.Length at {DirectoryName.Length} is not < 248")
+                Dim FullPathToTargetDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), DirectoryName)
+                Assert.True(FullPathToTargetDirectory.Length < 260, $"FullPathToTargetDirectory.Length at {FullPathToTargetDirectory.Length} is not < 260")
+
+                FileSystem.CreateDirectory(FullPathToTargetDirectory)
+                Assert.True(IO.Directory.Exists(FullPathToTargetDirectory))
+                If AreAllLongPathsAvailable() Then
+                    Dim VeryLongDirectoryName As String = New String("A"c, 250)
+                    Dim VeryLongFullPathToTargetDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), VeryLongDirectoryName, VeryLongDirectoryName)
+                    FileSystem.CreateDirectory(VeryLongFullPathToTargetDirectory)
+                    Assert.True(IO.Directory.Exists(VeryLongFullPathToTargetDirectory))
+                Else
+                    Dim VeryLongDirectoryName As String = New String("A"c, 250)
+                    Dim VeryLongFullPathToTargetDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), VeryLongDirectoryName, VeryLongDirectoryName)
+                    Assert.Throws(Of IO.PathTooLongException)(Sub() FileSystem.CreateDirectory(VeryLongFullPathToTargetDirectory))
+                End If
+            End Using
+
+        End Sub
```

Comment:
This test should not require cloning PathFeatures.cs. If the C# implementation of `AreAllLongPathsAvailable` cannot be used directly from VB (because PathFeatures.cs is typically a linked file?), please either implement this test in C# or simplify the test.

The test could be simplified by targeting non-Windows platforms only or by checking the platform when an exception is thrown:
```VB
Try
    FileSystem.CreateDirectory(path)
    Assert.True(Directory.Exists(path))
Catch e As PathTooLongException
    Assert.True(RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
End Try
```

Reply:
The C# version is not available because it is not accessible. It is only needed to detect a Windows platforms that don't support LongPathNames and in the test environment that is the system called "NETFX x86" one and I don't know how to uniquely identify it. It is the one test that fails below and my implementation of don't work to properly detect the reason. I am writing a simpiler one.

Reply:
Is the code snippet I suggested above sufficient for this test?

Reply:
I think this is sufficient for now. A better platform check can be added later if necessary.

Reply:
I have added to Issue #3387, it seems like this is know and being treated as documentation issues, I think it is a code bug.